deployment/model-testing/check_model_dependencies.ipynb

```diff
@@ -0,0 +1,204 @@
+    "\n",
+    "client = Client()"
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "# Example Model Class"
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "outputs": [],
+   "source": [
+    "class ExampleModel(VertaModelBase):\n",
+    "    def __init__(self, artifacts):\n",
+    "        pass\n",
+    "\n",
+    "    @verify_io\n",
+    "    def predict(self, input):\n",
+    "        df = pd.DataFrame(input)\n",
+    "        return df.iloc[:1]"
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "# Check Model Dependencies Manually"
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "outputs": [],
+   "source": [
+    "assumed_environment = Python(requirements=[\"numpy\"])  # incorrect package to support `pandas` dependency in model\n",
+    "\n",
+    "# Raises a warning by default\n",
+    "check_model_dependencies(model_cls=ExampleModel, environment=assumed_environment)\n",
+    "\n",
+    "# Raises an exception if `raise_for_missing=True`\n",
+    "check_model_dependencies(model_cls=ExampleModel, environment=assumed_environment, raise_for_missing=True)\n"
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "# Check Dependencies While Creating a Registered Model Version"
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "## Create a Registered Model"
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "outputs": [],
+   "source": [
+    "registered_model: RegisteredModel = client.get_or_create_registered_model(\n",
+    "    name=\"pandas_dependent_model_example\", labels=[\"examples\"])"
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "## Create a Registered Model Version\n",
+    "_Note the addition of the argument `check_model_dependencies=True` to trigger the dependency check._\n",
+    "\n",
+    "This is the equivalent of running:\n",
+    "```python\n",
+    "verta.registry.check_model_dependencies(model_cls=ExampleModel, environment=Python([]), raise_for_missing=True)\n",
+    "```"
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "outputs": [],
+   "source": [
+    "model_version: RegisteredModelVersion = registered_model.create_standard_model(\n",
+    "    model_cls=ExampleModel,\n",
+    "    environment=Python([]),\n",
+    "    code_dependencies=[],\n",
+    "    artifacts=None,\n",
+    "    model_api={\"predict\": {\"input\": \"dataframe\", \"output\": \"dataframe\"}},\n",
```

Comment:
I'd suggest leaving out arguments that aren't related to checking code dependencies. They're all optional, and they make it harder for a reader to tell what is relevant to the feature in question.

Reply:
dropped args